src/servant/tt_001/F84.c

```diff
@@ -57,7 +77,254 @@ Entity* func_us_80171568(Entity* self, s32 entityId) {
+            if (PLAYER.posX.i.hi >= self->posX.i.hi)
+                self->facingLeft = 1;
+            else
+                self->facingLeft = 0;
+        } else {
+            if (PLAYER.facingLeft == self->facingLeft) {
+                if (abs((s32)D_us_801737C4 - self->posX.i.hi) <= 0) {
+                    self->facingLeft = !PLAYER.facingLeft;
+
+                } else { // 3e0
+                    if (self->facingLeft && D_us_801737C4 < self->posX.i.hi) {
+                        if (PLAYER.facingLeft)
+                            temp_v0 = 0;
+                        else
+                            temp_v0 = 1;
+                        self->facingLeft = (u16)temp_v0;
+                    } else if (!self->facingLeft) {
+                        if (D_us_801737C4 > self->posX.i.hi) {
+                            if (PLAYER.facingLeft)
+                                temp_v0 = 0;
+                            else
+                                temp_v0 = 1;
+                            self->facingLeft = (u16)temp_v0;
+                        }
+                    }
+                }
+            } else {
+                if (self->facingLeft &&
+                    self->posX.i.hi - D_us_801737C4 >= 0x20) {
+                    self->facingLeft = PLAYER.facingLeft;
+                } else if (!self->facingLeft) {
+                    if (D_us_801737C4 - self->posX.i.hi >= 0x20) {
+                        self->facingLeft = PLAYER.facingLeft;
+                    }
+                }
+            }
+        }
+        func_us_80170F84(self, D_us_801737C4, D_us_801737C8);
+        self->posX.val += self->velocityX;
+        self->posY.val += self->velocityY;
+        if (D_8003C704 == 0) {
+            self->ext.ghost.unkA2 = func_us_80171284(self);
+            if (self->ext.ghost.unkA2) {
+                self->step++;
+                break;
+            }
+        }
+        break;
+    case 2:
+    case 3:
+        if (g_Player.unk0C & 0x01000001) {
+            self->step = 4;
+            self->ext.ghost.unk8C = 0;
+            break;
+        }
+        if (D_8003C704) {
+            self->step = 1;
+        }
+        if (!CheckEntityValid(self->ext.ghost.unkA2)) {
+            self->step = 1;
+            break;
+        }
+
+        D_us_801737C4 = self->ext.ghost.unkA2->posX.i.hi;
+        D_us_801737C8 = self->ext.ghost.unkA2->posY.i.hi;
+        if (self->velocityX > 0) {
+            self->facingLeft = 1;
+        }
+        if (self->velocityX < 0) {
+            self->facingLeft = 0;
+        }
+        D_us_801737CC =
+            func_us_80170F84(self, (s16)D_us_801737C4, D_us_801737C8);
+        if (self->step == 2) {
+            if (D_us_801737CC < 8) {
+                self->ext.ghost.unk86 = 0;
+                self->step += 1;
+            }
+        } else if (D_us_801737CC < 8) {
+            self->ext.ghost.unk86++;
+            temp_v1 = *(&D_us_80170508 + ((D_us_80173810.level / 10) * 3));
```

Comment:
Should be an array access

`D_us_80170508[((D_us_80173810.level / 10) * 3)]` maybe ?